src/extlibs/er_force_sim/src/amun/simulator/simball.cpp

```diff
@@ -69,31 +69,72 @@ SimBall::~SimBall()
     delete m_motionState;
 }
 
-void SimBall::begin()
+void SimBall::begin(bool robotCollision)
 {
     // custom implementation of rolling friction
-    const btVector3 p = m_body->getWorldTransform().getOrigin();
+    const btVector3 p        = m_body->getWorldTransform().getOrigin();
+    const btVector3 velocity = m_body->getLinearVelocity();
     if (p.z() < BALL_RADIUS * 1.1 * SIMULATOR_SCALE)
     {  // ball is on the ground
-        const btVector3 velocity = m_body->getLinearVelocity();
-        if (velocity.length() < 0.01 * SIMULATOR_SCALE)
+        BallState nextState = currentBallState;
+        bool isStationary   = velocity.length() < STATIONARY_BALL_SPEED * SIMULATOR_SCALE;
+        bool shouldRoll     = velocity.length() < rolling_speed * SIMULATOR_SCALE;
+        switch (currentBallState)
         {
-            // stop the ball if it is really slow
-            // -> the real ball snaps to a dimple
-            m_body->setLinearVelocity(btVector3(0, 0, 0));
+            case STATIONARY:
+                nextState = (robotCollision) ? ROBOT_COLLISION
+                                             : (isStationary) ? STATIONARY : SLIDING;
+                break;
+            case ROBOT_COLLISION:
+                nextState = (robotCollision) ? ROBOT_COLLISION
+                                             : (isStationary) ? STATIONARY : SLIDING;
+                break;
```

Comment:
seems like there's duplication in the cases, so if else may be better

Reply:
removed duplication